plugin/keyring_vault/vault_keyring.cc

```diff
@@ -47,7 +74,13 @@ int check_keyring_file_data(MYSQL_THD thd  MY_ATTRIBUTE((unused)),
 
   try
   {
-    boost::movelib::unique_ptr<IVault_curl> vault_curl(new Vault_curl(logger.get()));
+    if (reset_curl())
+    {
+      logger->log(MY_ERROR_LEVEL, "Cannot set keyring_vault_config_file");
+      mysql_rwlock_unlock(&LOCK_keyring);
+      return 1;
+    }
+    boost::movelib::unique_ptr<IVault_curl> vault_curl(new Vault_curl(logger.get(), curl));
     boost::movelib::unique_ptr<IVault_parser> vault_parser(new Vault_parser(logger.get()));
     IKeyring_io *keyring_io(new Vault_io(logger.get(), vault_curl.release(), vault_parser.release()));
```

Comment:
I have changed it, although Vault_io constructor cannot currently throw an exception. I guess it can in the future, so what you propose is a safer approach.